sympy/concrete/guess.py

```diff
@@ -0,0 +1,215 @@
+"""Various algorithms for helping identifying numbers and sequences."""
+from __future__ import print_function, division
+
+from sympy.utilities import public
+
+from sympy.core import Function, var
+from sympy.core.numbers import Zero
+from sympy import sympify, floor, sqrt
+from mpmath import pslq, sqrt, mp
+
+@public
+def find_simple_recurrence_vector(v, maxcoeff=1024):
+    """
+    This function is used internally by other functions from the
+    sympy.concrete.guess module. While most users may want to rather use the
+    function find_simple_recurrence when looking for recurrence relations
+    among rational numbers, the current function may still be useful when
+    some post-processing has to be done.
+
+    The function returns a vector of length n when a recurrence relation of
+    order n is detected in the sequence of rational numbers v.
+
+    If the returned vector has a length 1, then the returned value is also [0],
+    which means that no relation has been found.
+
+    Examples
+    ========
+
+    >>> from sympy.concrete.guess import find_simple_recurrence_vector
+    >>> from mpmath import fib
+    >>> find_simple_recurrence_vector( [ fib(k) for k in range(12) ] )
+    [1, -1, -1]
+
+    See also
+    ========
+
+    See the function sympy.concrete.guess.find_simple_recurrence which is more
+    user-friendly.
+
+    """
+    l = len(v)>>1
+
+    previous = mp.prec # save current precision
+    mp.prec = 128
+    b = [ sum( sqrt((l>>1)**2 + k)*v[-1-k-i] for k in range(l) )
+          for i in range(l) ]
+    p = pslq(   b,
+                maxcoeff = maxcoeff,
+                maxsteps = 128 + 4*l
+            )
+    mp.prec = previous # restore current precision
+    if p == None: return [0]
+
+    first, last = 0, l-1
+    while p[first]==0: first += 1
+    while p[last]==0:
+        last -= 1
+        if first == last: return [0] # TODO: probably never occuring
+    return p[first:last+1]
+
+@public
+def find_simple_recurrence(v, A=Function('a'), N=var('n'), maxcoeff=1024):
```

Comment:
You should never use `var()` in the library code, tests and examples/doctests. `var()` is magic and is reserved solely for interactive use. Use either `Symbol` constructor (as you used `Function`) or `symbols()`.